src/_pytest/config/argparsing.py

```diff
@@ -203,6 +211,18 @@ def addini(
         :py:func:`config.getini(name) <pytest.Config.getini>`.
         """
         assert type in (None, "string", "paths", "pathlist", "args", "linelist", "bool")
+        if default is notset:
+            if type is None:
+                default = ""
+            else:
+                if type in ["paths", "pathlist", "args", "linelist"]:
+                    default = []
+                elif type == "bool":
+                    default = False
+                else:
+                    # for string type
+                    default = ""
```

Comment:
Suggestion, move this code to a free function:

```suggestion
        if default is notset:
            default = get_ini_default_for_type(type)
```

And then we can easily unit-test this function, rather than using `pytester`.